### PR TITLE
Fix brain dict regression

### LINKED ISSUE #2200: Invariance "MANAGER.__dict__ == manager.AstroidManager.brain" no longer holds
### Steps to reproduce
Up until commit bbcc58bd52e7f295b77a8618b19b2364625590a2 the following equality would be guaranteed:

```
assert MANAGER.always_load_extensions == manager.AstroidManager.brain['always_load_extensions']
```

This apparently causes https://github.com/pylint-dev/pylint/issues/7433

### PR DESCRIPTION
**Pull request by knedlsepp**
## Type of Changes

This fixes issue #2200, which should fix https://github.com/pylint-dev/pylint/issues/7433.

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Closes #2200

### CODE REVIEW

**Comment by knedlsepp** [comment]
Put back as draft as this apparently breaks tests.

**Comment by knedlsepp** [comment]
@jacobtylerwalls I tried to implement your suggestion but with that I apparently run into other test failures. I could get pylint working again with:

```diff
--- a/astroid/manager.py
+++ b/astroid/manager.py
@@ -66,6 +66,7 @@ class AstroidManager:
 
     def __init__(self) -> None:
         # NOTE: cache entries are added by the [re]builder
+        self.__dict__ = AstroidManager.brain
         self.astroid_cache = AstroidManager.brain["astroid_cache"]
         self._mod_file_cache = AstroidManager.brain["_mod_file_cache"]
         self._failed_import_hooks = AstroidManager.brain["_failed_import_hooks"]
```

At the moment I'm stuck as I don't know enough about astroid to continue here.

**Comment by jacobtylerwalls** [comment]
Interesting. What happens if you revert the changes relating to the mutable values like `{}` and just introduce properties for manipulating the immutable ones like the booleans?

**Review by jacobtylerwalls — APPROVED** [review]
Thanks for identifying the issue and sending a patch. Welcome aboard!

File: tests/test_regrtest.py [hunk 1]
```diff
@@ -36,6 +36,12 @@ def tearDown(self) -> None:
         sys.path.pop(0)
         sys.path_importer_cache.pop(resources.find("data"), None)
 
+    def test_brain_dict(self) -> None:
```
[the remark this answers is not in the document]
Comment (jacobtylerwalls) [reply tests/test_regrtest.py:39]: It's hard to see at a glance why this test would even have a chance to fail: it's the setUp method fiddling with the value that was getting lost. Maybe a docstring or comment would help.

**Review by jacobtylerwalls — DISMISSED** [review]

**Review by jacobtylerwalls — DISMISSED** [review]

**Comment by DanielNoord** [comment]
@jacobtylerwalls Do you want to add typing to these signatures?

**Comment by jacobtylerwalls** [comment]
Nah, doesn't need to be done now. Doesn't provide much value anyway.

**Comment by jacobtylerwalls** [comment]
The missing coverage is worth adding, though!

**Comment by knedlsepp** [comment]
Thank you!

**Pull request merged**